tests/app/forms/validation/test_number_range_validator.py

```diff
@@ -95,6 +97,31 @@ def test_within_range_at_max(self):
         except ValidationError:
             self.fail('Valid integer raised ValidationError')
 
+    def test_answer_id_invalid_type(self):
+        answer = {
+            'max_value': {'answer_id': 'set-maximum-cat'},
+            'label': 'Range Test 10 to 20',
+            'mandatory': False,
+            'validation': {
+                'messages': {
+                    'INVALID_NUMBER': 'Please only enter whole numbers into the field.',
+                    'NUMBER_TOO_SMALL': 'The minimum value allowed is 10. Please correct your answer.',
+                    'NUMBER_TOO_LARGE': 'The maximum value allowed is 20. Please correct your answer.',
+                }
+            },
+            'id': 'test-range',
+            'type': 'Currency',
+        }
+        returned_error_messages = answer['validation']['messages']
+
+        with self.assertRaises(Exception) as ite:
+            get_field(answer, returned_error_messages, self.store, {}, None, False)
+
+            self.assertEqual(
+                str(ite.exception),
+                'answer: set-maximum-cat value: cat for answer id: test-range is not a valid number',
+            )
```

Comment:
I think all of the tests need to be re-organised to match the code changes. For example, this test is in the number range validator, but the value is resolved in the field handler.